### PR TITLE
Added Frontend AccountView

### PR DESCRIPTION
**Pull request by ProgramPhoenix**
adding the frontend account view

### CODE REVIEW

**Comment by Kr0nox** [comment]
The use of the rest interfaces should be restricted to views. They should not be used in components

**Comment by ProgramPhoenix** [comment]
why? that would be very bad, since now the save button needed to do an emit, so the view could call the rest interface. also we need the accountdetailseditcontainer at least twice, so it would be a lot of duplicated code

**Comment by c0derMo** [comment]
Saving in here also makes it easier to display the spinner on the saving button - otherwise, we'd have to use an additional prop

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    } else if (e.code === LoernwerkErrorCodes.FORBIDDEN) {
+      forbidden.value = true;
+    } else {
+      throw e;
+    }
+  } else {
+    throw e;
+  }
+} finally {
+  originalUser = { type: UserClass.REGULAR };
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: What is this for specifically?

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    default: false,
+  },
+  userid: {
+    type: Number,
+    required: false,
+    default: null,
+  },
+});
+
+const nameField = ref('');
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: We could pre-fill all of these if we set them after we query the account, correct?

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  userid: {
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: Missing doc

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    } else {
+      displayError.value = true;
+      disableInputShowSpinner.value = false;
+      return;
+    }
+  }
+  if (props.showadminview) {
+    updateUser.type = isAdmin.value ? UserClass.ADMIN : UserClass.REGULAR;
+  }
+  AccountRestInterface.updateAccount(updateUser);
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: missing await

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+/**
+ * deletes a account
+ */
+async function deleteAccount(): Promise<void> {
+  disableInputShowSpinner.value = true;
+  if (originalUser.id === null) {
+    console.log('user id is empty - should not happen');
+    return;
+  }
+  AccountRestInterface.deleteAccount(originalUser.id as number);
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: missing await

File: frontend/src/components/TextInputComponent.vue [hunk 6]
```diff
@@ -63,6 +63,11 @@ defineProps({
     required: false,
     default: false,
   },
+  startText: {
```
Comment (c0derMo) [inline frontend/src/components/TextInputComponent.vue]: missing doc

File: frontend/src/views/AccountView.vue [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<template>
+  <div class="w-full mt-auto mb-auto ml-3 mr-3">
+    <Suspense>
+      <AccountDetailsEditContainer :showadminview="true" />
```
Comment (c0derMo) [inline frontend/src/views/AccountView.vue]: We probably shouldnt have the admin view in the regular account editor

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+    }
+  } else {
+    throw e;
+  }
+} finally {
+  originalUser = { type: UserClass.REGULAR };
+}
+const isAdmin = ref(originalUser.type === UserClass.ADMIN);
+
+console.log(originalUser);
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: trailing console.log

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+      displayError.value = true;
+      disableInputShowSpinner.value = false;
+      return;
+    }
+  }
+  if (props.showadminview) {
+    updateUser.type = isAdmin.value ? UserClass.ADMIN : UserClass.REGULAR;
+  }
+  AccountRestInterface.updateAccount(updateUser);
+  disableInputShowSpinner.value = false;
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue:203]: Can we get some feedback that saving was successful?

**Review by c0derMo — CHANGES_REQUESTED** [review]

**Comment by Kr0nox** [comment]
> why? that would be very bad, since now the save button needed to do an emit, so the view could call the rest interface. also we need the accountdetailseditcontainer at least twice, so it would be a lot of duplicated code

For one our entire project s curently modeled like that.

The principle that components should be versetile and disconnected from the logic. You want to give a component data to display and manage, but you dont want that component to directly access and modify data at will

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 1 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: hmm This does not make any sense and will be removed soon

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 2 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: shouldn't matter, on an edit they are correctly filled. but if you are not changing something and they are empty (="") im not sending a request. But i guess it would be better to fill them and checking if they changed from the fetched user

File: frontend/src/views/AccountView.vue (hunk 7 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: yeah that would be bad

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 2 above)
Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Not sure if checking is even required, even if we send a request though nothing changed, I dont think it's that bad

**Comment by ProgramPhoenix** [comment]
i moved the request for the user in the view, but i still think that the place for saving should be in the component to avoid code duplication. we also never declared emits for the `AccountDetailsEditContainer ` in our Design

File: frontend/src/views/AccountView.vue (hunk 7 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: since i want to test the component also for Adminview this will stay until just before merge. But you could review again, since i did some major changes @c0derMo

File: dev.db
Comment (c0derMo) [inline dev.db:1]: This file shouldn't be part of the PR

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+const mailField = ref(props.user.mail);
+const pwField = ref('');
+const pwFieldControl = ref('');
+const disableInputShowSpinner = ref(false);
+const deleted = ref(false);
+const displayError = ref(false);
+const displaySuccess = ref(false);
+let originalUser = props.user as IUser;
+const isAdmin = ref(originalUser.type === UserClass.ADMIN);
+console.log('test');
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: console.log

File: frontend/src/restInterfaces/AccountRestInterface.ts [hunk 11]
```diff
@@ -65,6 +65,16 @@ export class AccountRestInterface extends BaseRestInterface {
   public static async getOwnAccount(): Promise<Partial<IUser>> {
     return await BaseRestInterface.get<Partial<IUser>>(this.account_path);
   }
+  /**
+   * Sends request to backend to get the account of the user with the given id
+   * @param id the id of the user
+   * @returns the requested user
+   */
+  public static async getAccount(id: number): Promise<Partial<IUser>> {
```
Comment (c0derMo) [inline frontend/src/restInterfaces/AccountRestInterface.ts]: This method doesn't even have a valid endpoint in the backend - I'd remove it here and implement it with the admin view, once that's done

**Review by c0derMo — CHANGES_REQUESTED** [review]
And the restinterface in components vs only in views discussion still stands - unsure what to do

File: frontend/src/restInterfaces/AccountRestInterface.ts (hunk 11 above)
Comment (ProgramPhoenix) [reply frontend/src/restInterfaces/AccountRestInterface.ts]: there should be one and that is :`accountRouter.get('/:ids', async (req, res) => { ...` (line 121 in accountrouter), it just didnt exited in the AccountRestInterface. But i dont need it anymore so i can remove it if you want

Comment (c0derMo) [reply frontend/src/restInterfaces/AccountRestInterface.ts]: That one has a method in the interface aswell, `getAccounts`, but that endpoint is only for querying usernames to ids, not full user objects

Comment (ProgramPhoenix) [reply frontend/src/restInterfaces/AccountRestInterface.ts]: kk i'll remove it

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 12]
```diff
@@ -0,0 +1,225 @@
+<template>
+  <div>
+    <ContainerComponent>
+      <template #Header>
+        <h1 class="underline text-xl">Daten ändern:</h1>
+      </template>
+
+      <template #default>
+        <div class="text-gray-400 p-1 text-xs">
+          Hinweis: Sie müssen nicht alle Felder ausfüllen, leere Felder werden
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: I dont think this is a good thing. 
Many users wont skip these texts escpecially since they are grayed out and by that seem "irrelevant".
So when the field is empty a uer would expect to be able to set the field to empty

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+              <TextInputComponent
+                :disabled="disableInputShowSpinner || deleted"
+                :hidden="true"
+                place-holder="Passwort wiederholen"
+                @input-changed="(val) => (pwFieldControl = val)"
+                :max-length="128"
+              />
+            </td>
+          </tr>
+          <tr :class="showadminview ? '' : 'collapse'">
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: use `v-if`

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+            <td>
+              <div>
+                <input
+                  :disabled="disableInputShowSpinner || deleted"
+                  type="checkbox"
+                  class="cursor-pointer m-1"
+                  v-model="isAdmin"
+                  :checked="isAdmin"
+                />
+                <label class="cursor-pointer" @click="toggleCheckBox">
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: ambigous name with so many inputs

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+  if (deleted.value) {
+    return;
+  }
+  displaySuccess.value = false;
+  displayError.value = false;
+  disableInputShowSpinner.value = true;
+
+  const updateUser: Partial<IUser> = { id: originalUser.id };
+
+  if (nameField.value !== '') {
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: I dont think we should check for them to be empty in this way. The user should have the choice to do so. It should display an error when trying to save then

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="updateInformation()"
+          >
+            Speichern
+          </ButtonComponent>
+        </div>
+        <div :class="showadminview ? 'flex items-center pt-1' : 'collapse'">
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: use v-if

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+    } else {
+      displayError.value = true;
+      disableInputShowSpinner.value = false;
+      return;
+    }
+  }
+  if (props.showadminview) {
+    updateUser.type = isAdmin.value ? UserClass.ADMIN : UserClass.REGULAR;
+  }
+  await AccountRestInterface.updateAccount(updateUser);
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: I think we should do a check, whether the mail is correct

Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: I still think the rest interfaces have no place here.
Vue Documentation: 
> Components allow us to split the UI into independent and reusable pieces, and think about each piece in isolation.

This would not be the case, since it is heavily tied to this function of the rest interfaces

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+  deleted.value = true;
+}
+/**
+ * reseting this component
+ */
+async function reset(): Promise<void> {
+  if (deleted.value) {
+    return;
+  }
+  window.location.reload();
```
Comment (Kr0nox) [inline frontend/src/components/AccountDetailsEditContainer.vue]: This isnt a good solution. Reloading should be avoided in a single page application. Use the given user and reset all fields

File: frontend/src/views/AccountView.vue [hunk 19]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <div class="flex-grow text-center">
+    <div class="text-green-500 italic" v-if="showerror">
+      Account nicht verfügbar
+    </div>
+  </div>
+  <div class="w-full mt-auto mb-auto ml-3 mr-3">
+    <AccountDetailsEditContainer
+      :showadminview="true"
```
Comment (Kr0nox) [inline frontend/src/views/AccountView.vue]: Shouldnt this be false in the account view?
Or at least for non admins

File: frontend/src/views/AccountView.vue [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+import { ref } from 'vue';
+
+const showerror = ref(false);
+let originalUser: Partial<IUser>;
+try {
+  originalUser = await AccountRestInterface.getOwnAccount();
+  console.log(originalUser);
+} catch (e) {
+  if (e instanceof LoernwerkError) {
+    if (e.code === LoernwerkErrorCodes.UNAUTHORIZED) {
```
Comment (Kr0nox) [inline frontend/src/views/AccountView.vue]: We could implement a utiliy function for this sort of error handing, since some reactions might be used on all pages

**Review by Kr0nox — CHANGES_REQUESTED** [review]

File: frontend/src/views/AccountView.vue (hunk 19 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: see above

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 14 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: we only have one checkbox

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 18 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: i dont see a better solution, since the last reset option i found is for vue2 (see: https://github.com/vuejs/vue/issues/702)

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 14 above)
Comment (Kr0nox) [reply frontend/src/components/AccountDetailsEditContainer.vue]: I know, what I ment is in case more get added for some reason. And even now it is a very undescribtive name

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 17 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: since the button should always do the same in all components, namely save the user, they should not have the option to get rebind i.e. to a deleting call.
also taking logic out of the component would lead to copy paste, what would be much worse than violating a dokumentation(which also more sounds like a hint or recommendation) , because it ~~could~~ always lead to massive inconsistencies

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 15 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: hm since we dont enforce the user to set a new password when changing the mail or username (or do we want to?), it would be the consistent way to handle empty fields

File: frontend/src/views/AccountView.vue (hunk 20 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: im not sure if we can get a advantage from this, because we may need to do diverse things on errors, the only common action (i currently could imagine) would be a redirect on a unauthorized error

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 12 above)
Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Since fields are pre-filled already anyways, I dont think this notice is needed anymore

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 15 above)
Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: But since we're pre-filling, we'd have to check them against the original user, not empty

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 17 above)
Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: regex or something would work I think, just a basic check

Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: I do agree with the point that e.g. a save button won't logically be bound to anything other than our save function, in my opinion, the container cant logically work on its own without the save/delete functionality

Also, if we need to change the signature of our save function, we'd only have to change it in this location, instead of rechecking every location for the save function anywhere, and fixing it

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 18 above)
Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: I think the better option here is just to reset all the fields to the values of the originalUser and empty the password field, since we still got the reference to it in the user prop

Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: i still dont see how that would work since the input field is capsulated in a component. You may could push it or write how the code would look like?

Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Shouldnt something like
`nameField.value = props.user.name` (similar for the others)
work?

Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: that would work in javascript (stackoverflow: `document.getElementById('MyId').value = newVal`) right? in vue according to my research i would need to have access to the v-model variable and set these to the given value, but i cant access these in the component (i know that vue compiles to javascript, but i cant access the document variable in the `<script>` section)

Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Ah, I see, because we can't override the value of TextInputComponents after we created them, at least right now - however, we could make our TextInputComponent compatible with v-model, but that'd change a fair bit of our existing work, since all TextInputComponents would need to be changed, although they'd fit in better with "regular vue" after I'd imagine - Opinions? (Also @Kr0nox)

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 17 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: i cant push regex cause of the no-useless-escape rule from lint. (`155:19  error  Unnecessary escape character: \.  no-useless-escape`)

Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Do you need the escape? What regex are you trying to use?

Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: in regex . ist one character, but we need the . i.e in .com or max.musterman@mail.de

Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: ~~Makes sense, I guess the easiest way to circumvent that, would just disabling `no-useless-escape` or at least allowing escaping dots, I'll see what I can find in the config options~~ EDIT: See below

Comment (c0derMo) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Actually, I think this behavior might even be correct, since you need to escape the `\` before the `.` in js, otherwise it'll be interpreted as a JS-escape-sequence instead of a regex one
Suggestions to fix this would include either using slashes to declare your regex (like `const regex = /REGEX_HERE/`) or escaping the `\.` to `\\.` in the RegExp constructor. At least these suggestions made it through the lint for me

Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: `/REGEX_HERE/` did not seem to work, replacing `\.` with `\\.` fixed the problem. just a last thing about the regex: the original regex i found allowed something like `max.mustermann@musterhausen` and did not required a tld. I changed the regex to require one. What do you think? should we allow mails without tld?

Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Status update: We decided to not allow hostname. Resolving this Conversation

Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Status update: We decided to allow restinterface calls in compnents. Resolving this conversation

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 18 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Status update: We decided to make the TextInputComponent v-model compatible. Updating this soon:tm:

File: frontend/src/views/AccountView.vue (hunk 20 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: Status update: We did not talk about this, but i think i can come up with a shorter solution soon:tm:

File: frontend/src/components/AccountDetailsEditContainer.vue (hunk 18 above)
Comment (ProgramPhoenix) [reply frontend/src/components/AccountDetailsEditContainer.vue]: Status update: v-model is complete, this is now working without reloading the page

File: frontend/src/views/AccountView.vue (hunk 20 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: Made this a bit shorter. I would agree to do a utility function for the following part, since it would be occure in more classes: 
```ts
if (e instanceof LoernwerkError) {
 if (e.code === LoernwerkErrorCodes.UNAUTHORIZED) {
  router.push('LogIn');
 }
}
```
where should it be placed?

**Comment by ProgramPhoenix** [comment]
The PR is now ready for review. Please review again @c0derMo @Kr0nox  @loernwerk/active

File: frontend/src/components/TextInputComponent.vue [hunk 21]
```diff
@@ -63,6 +64,10 @@ defineProps({
     required: false,
     default: false,
   },
+  /**
+   *
```
Comment (c0derMo) [inline frontend/src/components/TextInputComponent.vue]: Missing documentation

File: frontend/src/components/TextInputComponent.vue [hunk 22]
```diff
@@ -6,9 +6,10 @@
       :maxLength="maxLength"
       :placeholder="placeHolder"
       :disabled="disabled"
-      v-model="inputText"
+      :value="props.input"
```
Comment (c0derMo) [inline frontend/src/components/TextInputComponent.vue]: Is this still needed?

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 23]
```diff
@@ -0,0 +1,232 @@
...
+              />
+            </td>
+          </tr>
+          <tr>
+            <td class="p-1">E-mail:</td>
+            <td class="p-1">
+              <TextInputComponent
+                :disabled="disableInputShowSpinner || deleted"
+                place-holder="E-mail"
+                :startText="originalUser.mail"
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: Is this still needed with the change to v-model?

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 24]
```diff
@@ -0,0 +1,232 @@
...
+
+      <template #default>
+        <table class="w-full">
+          <tr>
+            <td class="p-1">Nutzername:</td>
+            <td class="p-1 w-2/3">
+              <TextInputComponent
+                :disabled="disableInputShowSpinner || deleted"
+                place-holder="Nutzername"
+                :startText="originalUser.name"
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: Is this still needed with the change to v-model?

File: frontend/src/components/TextInputComponent.vue [hunk 25]
```diff
@@ -63,6 +64,10 @@ defineProps({
     required: false,
     default: false,
   },
+  /**
+   *
+   */
+  input: String,
```
Comment (c0derMo) [inline frontend/src/components/TextInputComponent.vue]: shouldn't we rename this to `modelValue` so we can just use `v-model` in other components instead of `v-model:input`?

File: frontend/src/views/LogInView.vue [hunk 26]
```diff
@@ -88,7 +88,7 @@ function disableCheckBox(): void {
 async function checkLogIn(): Promise<void> {
   disableInputShowSpinner.value = true;
   displayError.value = false;
-
+  console.log(mailField.value);
```
Comment (c0derMo) [inline frontend/src/views/LogInView.vue]: console.log

**Review by c0derMo — CHANGES_REQUESTED** [review]

File: frontend/src/components/TextInputComponent.vue (hunk 25 above)
Comment (ProgramPhoenix) [reply frontend/src/components/TextInputComponent.vue]: that didn't worked for me, i needed to write then `v-model:modelValue`, i will try it again, may one of my other problem, that i fixed in meantime, was the problem

File: frontend/src/components/TextInputComponent.vue (hunk 22 above)
Comment (ProgramPhoenix) [reply frontend/src/components/TextInputComponent.vue]: yes i think so, if i understand it right v-model will replaced with this 
```ts
<input
  :value="inputRef"
  @input="inputRef= $event.target.value"
/>
```
and i think the input is loaded before the reference is set, but however if i remove it, it doesnt work

File: frontend/src/components/TextInputComponent.vue (hunk 25 above)
Comment (ProgramPhoenix) [reply frontend/src/components/TextInputComponent.vue]: works actually now

File: frontend/src/components/TextInputComponent.vue [hunk 27]
```diff
@@ -6,9 +6,10 @@
       :maxLength="maxLength"
       :placeholder="placeHolder"
       :disabled="disabled"
-      v-model="inputText"
+      :value="props.modelValue"
```
Comment (c0derMo) [inline frontend/src/components/TextInputComponent.vue]: I dont think this is needed anymore, from my testing

File: frontend/src/views/AccountView.vue (hunk 20 above)
Comment (c0derMo) [reply frontend/src/views/AccountView.vue]: Update: The NavigationBar will already catch not-logged-in users, since it does need to query their data to see if it should display the admin button and will redirect to login if 401 is encountered, so I'm not sure if we need to still detect it everywhere

File: frontend/src/components/AccountDetailsEditContainer.vue [hunk 28]
```diff
@@ -0,0 +1,230 @@
...
+          </tr>
+          <tr>
+            <td class="p-1">Passwort wiederholen:</td>
+            <td class="p-1">
+              <TextInputComponent
+                :disabled="disableInputShowSpinner || deleted"
+                :hidden="true"
+                place-holder="Passwort wiederholen"
+                :max-length="128"
+                v-model="pwField"
```
Comment (c0derMo) [inline frontend/src/components/AccountDetailsEditContainer.vue]: this should presumably be `pwFieldControl`

**Review by c0derMo — CHANGES_REQUESTED** [review]
small, hopefully the last changes

File: frontend/src/views/AccountView.vue (hunk 20 above)
Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: that's great i will remove it

File: frontend/src/components/TextInputComponent.vue (hunk 27 above)
Comment (ProgramPhoenix) [reply frontend/src/components/TextInputComponent.vue]: hm, thats right i guess the hot reload from vite my not work for this right

**Comment by ProgramPhoenix** [comment]
everything resolved, pls review again @c0derMo  @loernwerk/active

File: frontend/src/views/AccountView.vue [hunk 29]
```diff
@@ -1,35 +1,14 @@
...
-  originalUser = await AccountRestInterface.getOwnAccount();
-} catch (e) {
-  if (e instanceof LoernwerkError) {
-    if (e.code === LoernwerkErrorCodes.UNAUTHORIZED) {
-      router.push('LogIn');
-    }
-  }
-  showerror.value = true;
-}
+originalUser = await AccountRestInterface.getOwnAccount();
```
Comment (c0derMo) [inline frontend/src/views/AccountView.vue]: Sorry for the misunderstanding, I'd still keep the try-catch, just in cases of other errors (e.g. account being deleted while logged in or something), just remove the `if (e instanceof LoernwerkError) { ... }` part, and only have the error message

Comment (ProgramPhoenix) [reply frontend/src/views/AccountView.vue]: ok now again :joy:

**Review by c0derMo — APPROVED** [review]

**Review by Kr0nox — APPROVED** [review]
